File: backend/src/main/java/com/eoullim_backend/dto/PostDTO.java
package com.eoullim_backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PostDTO {
    private Long id;
    private Long userId;
    private String title;
    private String content;
    private Integer viewCount;
    private Integer likeCount;
    private Integer commentCount; // 댓글 수 추가
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: backend-nestjs/src/posts/dto/post.dto.ts
import { IsString, IsNotEmpty, IsNumber, IsOptional, MaxLength, MinLength } from 'class-validator';

export class PostDTO {
  id: number;
  userId: number;
  title: string;
  content: string;
  viewCount: number;
  likeCount: number;
  commentCount: number; // 댓글 수 포함
  createdAt: Date;
  updatedAt: Date;
}

export class PostRequestDTO {
  @IsNumber({}, { message: '사용자 ID는 숫자여야 합니다.' })
  @IsNotEmpty({ message: '사용자 ID는 필수입니다.' })
  userId: number;

  @IsString({ message: '제목은 문자열이어야 합니다.' })
  @IsNotEmpty({ message: '제목은 필수입니다.' })
  @MinLength(1, { message: '제목은 최소 1자 이상이어야 합니다.' })
  @MaxLength(255, { message: '제목은 최대 255자까지 가능합니다.' })
  title: string;

  @IsString({ message: '내용은 문자열이어야 합니다.' })
  @IsNotEmpty({ message: '내용은 필수입니다.' })
  @MinLength(1, { message: '내용은 최소 1자 이상이어야 합니다.' })
  content: string;
}
